### DISCUSSION TITLE
DB Migrations

### DISCUSSION THREAD

**Discussion by m4tta** [nextauthjs/next-auth#1308] [Help]
Are we expected to run the SQL provided here https://next-auth.js.org/schemas/postgres or is there a proper way to generate typeorm migrations for the entities used in next-auth?

**Comment by Manubi** [comment]
Hi @m4tta,
right now there are database adapters for TepeORM and Prisma2. 
[documentation adapters](https://next-auth.js.org/schemas/adapters)

What they do is, they basically establish a database connection to your database and you can run your typeORM/prsima2 cli commands with it. But even with an adapter afaik the recommended way is to run the SQL and then inspect the schema with the used ORM.

I would also look for for a more flexible way to handle the connection between auth and database model, as right now the field names are also pretty much fixed. You can extend the model but you can't map a next-auth field eg. user.name (next-auth model) to user.displayname (your existing database field name) [documentation custom models](https://next-auth.js.org/tutorials/typeorm-custom-models)

Hope it helps.
